### PR TITLE
Add a delay option to mediarecorder

### PR DESCRIPTION
**Pull request by RedGuy12**
Resolves https://discord.com/channels/806602307750985799/806609030935740477/892989006927773726

### Changes

Add a delay option to mediarecorder

### Reason for changes

bc

### Tests

yes

### CODE REVIEW

**Comment by PoIygon** [comment]
You haven’t got any proof that you have done tests except for “yes”

**Comment by RedGuy12** [comment]
It works...


________________________________

File: addons/mediarecorder/userscript.js [hunk 1]
```diff
@@ -63,6 +63,24 @@ export default async ({ addon, console, msg }) => {
       recordOptionSeconds.appendChild(recordOptionSecondsInput);
       recordOptionInner.appendChild(recordOptionSeconds);
 
+      // Delay
+      const recordOptionDelay = document.createElement("p");
+      const recordOptionDelayInput = Object.assign(document.createElement("input"), {
+        type: "number",
+        min: 1,
```
Comment (WorldLanguages) [inline addons/mediarecorder/userscript.js:70]: Min should be 0

File: addons/mediarecorder/userscript.js [hunk 2]
```diff
@@ -298,7 +316,21 @@ export default async ({ addon, console, msg }) => {
         stopSignFunc = () => stopRecording();
         vm.runtime.once("PROJECT_STOP_ALL", stopSignFunc);
       }
-      recorder.start(1000);
+
+      // Delay
+      const delay = opts.delay || 0;
+      const roundedDelay = Math.floor(delay);
+      for (let index = 0; index < roundedDelay; index++) {
+        console.log(roundedDelay - index);
```
Comment (WorldLanguages) [inline addons/mediarecorder/userscript.js:324]: Remove console.log

File: addons/mediarecorder/userscript.js [hunk 3]
```diff
@@ -298,7 +316,21 @@ export default async ({ addon, console, msg }) => {
...
+        recordElem.textContent = msg("starting-in", { secs: roundedDelay - index });
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      }
+      console.log((delay - roundedDelay) * 1000);
```
Comment (WorldLanguages) [inline addons/mediarecorder/userscript.js:328]: Remove console.log

File: addons/mediarecorder/userscript.js [hunk 4]
```diff
@@ -298,7 +316,21 @@ export default async ({ addon, console, msg }) => {
...
       }
-      recorder.start(1000);
+
+      // Delay
+      const delay = opts.delay || 0;
+      const roundedDelay = Math.floor(delay);
+      for (let index = 0; index < roundedDelay; index++) {
+        console.log(roundedDelay - index);
+        recordElem.textContent = msg("starting-in", { secs: roundedDelay - index });
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (WorldLanguages) [inline addons/mediarecorder/userscript.js:326]: Maybe waiting 950ms would be more accurate overall.

File: addons/mediarecorder/userscript.js (hunk 1 above)
Comment (WorldLanguages) [reply addons/mediarecorder/userscript.js:70]: It would also be better to make this input smaller than duration, since it's less important

**Comment by WorldLanguages** [comment]
TODO: after modal is done, modify `margin-top` of `.mediaRecorderPopup` so that it stays centered vertically

File: addons/mediarecorder/userscript.js (hunk 1 above)
Comment (RedGuy12) [reply addons/mediarecorder/userscript.js]: Fixed min

I feel like making the input a different size would make it look off

File: addons/mediarecorder/userscript.js (hunk 4 above)
Comment (RedGuy12) [reply addons/mediarecorder/userscript.js:325]: Why?

Comment (WorldLanguages) [reply addons/mediarecorder/userscript.js:325]: Browsers rarely take exactly the amount of milliseconds you specify, it's often more than that

**Review by apple502j — APPROVED** [review]

**Pull request merged**